web_src/js/features/comp/ReactionSelector.js

```diff
@@ -1,38 +1,45 @@
 import $ from 'jquery';
 import {POST} from '../../modules/fetch.js';
 
-export function initCompReactionSelector($parent) {
-  $parent.find(`.select-reaction .item.reaction, .comment-reaction-button`).on('click', async function (e) {
-    e.preventDefault();
+export function initCompReactionSelector() {
+  const containers = document.querySelectorAll('.comment-list, .diff-file-body');
+  if (!containers.length) return;
 
-    if (this.classList.contains('disabled')) return;
+  for (const container of containers) {
+    container.addEventListener('click', async (e) => {
+      const item = e.target.matches('.item.reaction') ? e.target : e.target.closest('.item.reaction');
+      const button = e.target.matches('.comment-reaction-button') ? e.target : e.target.closest('.comment-reaction-button');
+      if (!item && !button) return;
+      e.preventDefault();
 
-    const actionUrl = this.closest('[data-action-url]')?.getAttribute('data-action-url');
-    const reactionContent = this.getAttribute('data-reaction-content');
-    const hasReacted = this.closest('.ui.segment.reactions')?.querySelector(`a[data-reaction-content="${reactionContent}"]`)?.getAttribute('data-has-reacted') === 'true';
+      const target = item || button;
+      if (target.classList.contains('disabled')) return;
 
-    const res = await POST(`${actionUrl}/${hasReacted ? 'unreact' : 'react'}`, {
-      data: new URLSearchParams({content: reactionContent}),
-    });
+      const actionUrl = target.closest('[data-action-url]')?.getAttribute('data-action-url');
```

Comment:
Most `?` should be removed. For example, if no "actionUrl", the following logic doesn't make sense.

It shouldn't tolerate such non-existing elements.

Reply:
Agree, better to be strict and throw there.

Reply:
That change actually made it throw now on the bug below. Nice :)

```
Uncaught (in promise) TypeError: can't access property "querySelector", target.closest(...) is null
```